addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -2458,7 +2460,9 @@ var FieldMany2ManyTags = AbstractField.extend({
         ev.stopPropagation();
         var newValue = ev.data.changes[this.name];
         if (newValue) {
-            this._addTag(newValue);
+            this._addTag(newValue)
+                .then(ev.data.onSuccess || function () {})
+                .guardedCatch(ev.data.onFailure || function () {});
             this.many2one.reinitialize(false);
```

Comment:
should not you wait for the add tag operation to complete before reinitializing?